crates/sozo/ops/src/account.rs

```diff
@@ -263,49 +308,20 @@ pub async fn deploy(
                 })?
                 .overall_fee;
 
-            let estimated_fee_with_buffer = (estimated_fee * felt!("3")).floor_div(felt!("2"));
-
-            if fee_setting.is_estimate_only() {
-                println!("{} ETH", format!("{}", estimated_fee.to_big_decimal(18)).bright_yellow(),);
-                return Ok(());
-            }
-
-            MaxFeeType::Estimated {
-                estimate: estimated_fee,
-                estimate_with_buffer: estimated_fee_with_buffer,
-            }
+            println!("{} ETH", format!("{}", estimated_fee.to_big_decimal(18)).bright_yellow());
+            Ok(())
+        }
+        TxnAction::Simulate => {
+            simulate_account_deploy(&account_deployment).await?;
+            Ok(())
         }
-    };
-
-    let account_deployment = match nonce {
-        Some(nonce) => account_deployment.nonce(nonce),
-        None => account_deployment,
-    };
-    let account_deployment = account_deployment.max_fee(max_fee.max_fee());
-
-    if simulate {
-        simulate_account_deploy(&account_deployment).await?;
-        Ok(())
-    } else {
-        do_account_deploy(
-            max_fee,
-            target_deployment_address,
-            no_confirmation,
-            account_deployment,
-            &provider,
-            poll_interval,
-            &mut account,
-        )
-        .await?;
-
-        write_account_to_file(file, account)?;
-
-        Ok(())
     }
 }
 
+#[allow(clippy::too_many_arguments)]
 async fn do_account_deploy(
     max_fee: MaxFeeType,
+    txn_config: TxnConfig,
```

Comment:
We still need the `MaxFeeType` with `TxnConfig`?

Reply:
yes since for account deployment we need to prompt users based on this info, for other subcommands its usually abstracted away by `starknet-rs`